src/main/java/duke/parser/Parser.java

```diff
@@ -0,0 +1,251 @@
+                } catch (DateTimeParseException e) {
+                    continue;
+                }
+                date = readout.atTime(0, 0);
+                if (date.isBefore(now)) {
+                    date = date.plusYears(1);
+                }
+            }
+
+            // All parameters supplied
+            else if (i <= 18) {
+                try {
+                    date = LocalDateTime.parse(input, fmt);
+                } catch (DateTimeParseException e) {
+                    continue;
+                }
+            }
+
+            // No year supplied
+            else {
+                try {
+                    date = LocalDateTime.parse(inputAddYear, fmt);
+                } catch (DateTimeParseException e) {
+                    continue;
+                }
+                if (date.isBefore(now)) {
+                    date = date.plusYears(1);
+                }
+            }
+
+            // Otherwise fully-supplied
+            return date;
+        }
+
+        // None of the date formats parsed
+        throw new DukeExceptionIllegalArgument("Datetime format should adhere to the following format:"
+                + "\n- Time delimited by ':', e.g. 9:30"
+                + "\n- Date delimited by '/' or ' ', e.g. 9/1, 9 Jan"
+                + "\n- Date and time separated by a single space ' '");
+    }
+
+    /**
+     * Returns formatted string based on supplied datetime.
+     *
+     * If task falls on current year but has not expired, no year will be inserted.
+     * Tasks falling midnight is considered not expired on the same day.
+     * If task has time specified other than midnight, the time will be inserted.
+     *
+     * @param dt LocalDateTime.
+     * @return Formatted datetime string.
+     */
+    public static String formatDate(LocalDateTime dt) {
+        LocalDateTime now = LocalDateTime.now();
+        if (dt.getHour() == 0 && dt.getMinute() == 0) {
+            if (dt.getYear() == now.getYear() && now.isBefore(dt.plusDays(1))) {
+                return dt.format(READABLE_DATE_FORMAT);
+            } else {
+                return dt.format(READABLE_DATE_YEAR_FORMAT);
+            }
+        } else {
+            if (dt.getYear() == now.getYear() && now.isBefore(dt)) {
+                return dt.format(READABLE_DATETIME_FORMAT);
+            } else {
+                return dt.format(READABLE_DATETIME_YEAR_FORMAT);
+            }
+        }
+    }
+
+    /**
+     * Returns formatted string as complete datetime.
+     *
+     * @param dt LocalDateTime.
+     * @return Formatted datetime string.
+     */
+    public static String formatDateFull(LocalDateTime dt) {
+        return dt.format(READABLE_DATETIME_YEAR_FORMAT);
+    }
+
+    /**
+     * Returns formatted string in ISO format, i.e. 'YYYY-MM-DD hh:mm'
+     *
+     * @param dt LocalDateTime.
+     * @return Formatted datetime string.
+     */
+    public static String formatDateISO(LocalDateTime dt) {
```

Comment:
Should the acronym ISO be lowercase (formatDateIso)?

Reply:
Yep, great catch, thanks!